.github/workflows/build-electron.yml

```diff
@@ -102,6 +102,13 @@ jobs:
       if: matrix.os != 'macos-latest'
       run: npm run test-electron-app
 
+    - name: Upload app logging from puppeteer to github
+      uses: actions/upload-artifact@v2.2.1
+      if: matrix.os != 'macos-latest' && always()
```

Comment:
What's the `always()` doing in this case? I read that it returns `true` even if cancelled.

Reply:
Making sure this step runs even if the test in the prior step failed, since that's when the logs are most important.